ibis/pandas/udf.py

```diff
@@ -532,23 +533,32 @@ def execute_udaf_node_groupby(op, *args, **kwargs):
                 # repeating it until all groups are exhausted.
                 aggcontext = kwargs.pop('aggcontext', None)
                 assert aggcontext is not None, 'aggcontext is None'
-                iters = (
-                    (data for _, data in arg)
-                    if isinstance(arg, SeriesGroupBy)
-                    else itertools.repeat(arg)
-                    for arg in args[1:]
-                )
-                funcsig = signature(func)
 
-                def aggregator(first, *rest, **kwargs):
-                    # map(next, *rest) gets the inputs for the next group
-                    # TODO: might be inefficient to do this on every call
-                    args, kwargs = arguments_from_signature(
-                        funcsig, first, *map(next, rest), **kwargs
+                if isinstance(aggcontext, Window):
+                    # Call the func differently for Window because of
+                    # the custom rolling logic.
+                    result = aggcontext.agg(args[0], func, *args, **kwargs)
+                else:
+                    iters = (
+                        (data for _, data in arg)
+                        if isinstance(arg, SeriesGroupBy)
+                        else itertools.repeat(arg)
+                        for arg in args[1:]
+                    )
+                    funcsig = signature(func)
+
+                    def aggregator(first, *rest, **kwargs):
```

Comment:
it would be nice to unify the calling convention for aggregate as well here to be more like window

maybe out of scope but would be cleaner i think as it’s conceptually the same

Reply:
Yeah I agree. I added a TODO for now. Unifying them probably needs some more refactoring,

Reply:
ok